internal/configurations/configurations.go

```diff
@@ -1,4 +1,11 @@
 // Package configurations encapsulates all the functionality around Epinio configurations
+// A Configuration is essentially a Secret with some Epinio specific labels.
+// This allows us to use any Secret as a Configuration as long as someone labels
+// it as such. In the future, we will use this to expose secrets created by Service
+// helm charts as Configurations (https://github.com/epinio/epinio/issues/1281).
+// Since we don't control the name of the produced secret in that case, we keep
+// the name of the Configuration in a label instead of just using the Secret's name,
+// to be able to have predictable Configuration names when a Service is provisioned.
```

Comment:
```suggestion
// A Configuration is essentially a Secret with some Epinio specific labels.
// This allows us to use any Secret as a Configuration as long as someone labels
// it as such. In the future, we will use this to expose secrets created by Service
// helm charts as Configurations (https://github.com/epinio/epinio/issues/1281).
// Since we don't control the name of the produced secret in that case, we will
// need some method to tie a Configuration to a Service. This can be solved with
// some labels on the Configuration or the Service instance resource.
```